### PR TITLE
[BugFix] Apply the request timeouts to the pooled Poco session

### PR DESCRIPTION
**Pull request by luohaha**
## Why I'm doing:

`PocoHttpClient` computes send and receive timeouts from the SDK's `ClientConfiguration` and
hands them to `getSession()`, which uses only `connection_timeout` — to bound how long a caller
waits for a session from the pool — and returns without applying the rest:

```cpp
// be/src/platform/aws/poco_common.cpp
auto session = pool->get(timeouts.connection_timeout.totalMicroseconds());
session->attachSessionData({});
return session;                       // send/receive timeouts dropped here
```

`setTimeouts()` is defined right above it and **has no caller anywhere in the tree**:

```
$ grep -rn 'setTimeouts' be/src/
be/src/platform/aws/poco_common.cpp:37:  void setTimeouts(...)     <- definition
be/src/platform/aws/poco_common.h:53:                              <- declaration
                                                                   <- no call site
```

So `object_storage_request_timeout_ms` is silently ineffective whenever this client is in use,
and a read that stops receiving waits out Poco's own default instead. Since
`SetHttpClientFactory()` is global and the BE sets `skip_aws_init_api = true` for starlet, this
covers both the shared-data lake read path and the BE's own `fs_s3`.

### What it looks like in production

Measured on a 3-node shared-data cluster (AWS, us-west-2) over cold SSB SF100 and ClickBench
scans. With the Poco client on, **7 of 522 queries (1.3%) hung, and every one of them landed at
58.6 ± 1.4 s** — a spread far too tight for network variance, and the signature of a single fixed
timeout. A query that normally takes 5 s takes 65 s when it hits this.

A capture taken while a query was still hung shows the scan thread parked in `recvfrom` with TCP
in perfect health, so this is the client waiting rather than the network failing:

```
tid=4307 comm=pip_con_scan_co  wchan=wait_woken  syscall=45 (recvfrom)

ESTAB  Recv-Q=0  Send-Q=0  rto:223  (no retransmits)
bytes_sent:23745  bytes_acked:23746        every byte we sent was acknowledged
bytes_received:25,174,087                  response body 25 MB in
lastsnd:31043  lastrcv:31014               peer silent for 31 s
rcv_space:550982  rcv_ssthresh:2165690     receive window wide open
```

## What I'm doing:

**1. Applying the per-request timeouts to the session** before handing it out, through a separate
helper rather than the existing `setTimeouts()`. That helper deliberately differs in two ways:

- **Keep-alive is left as the pool set it.** `ConnectionTimeouts` default-initializes
  `http_keep_alive_timeout` to zero and `PocoHttpClient` never sets it, so calling `setTimeouts()`
  would push a zero keep-alive onto a pooled session — working against the pool, which exists to
  reuse connections.
- **A request with no timeout restores Poco's default rather than skipping the call.** Sessions are
  pooled per endpoint, not per client, and clients sharing an endpoint do not share a timeout: a
  RENAME_FILE client carries `object_storage_rename_file_request_timeout_ms` (30 s by default)
  while an ordinary read carries `object_storage_request_timeout_ms`. Skipping would let a read
  inherit the rename client's 30 s. The default is read from a freshly constructed session rather
  than hardcoded, so it cannot drift from the bundled Poco version.

**2. Defaulting `object_storage_request_timeout_ms` to 10 s.** Unset, a stalled read waits out
whatever the client defaults to — 60 s on the Poco path, and *nothing at all* on curl, where the
SDK's own `requestTimeoutMs = 0` disables the low-speed check outright.

10 s is a stall detector, not a deadline. For curl it is the low speed time: the transfer must sit
below 1 byte/s for the whole window before the client gives up, so a slow-but-moving transfer is
never cut off however long it runs. For Poco it is the socket send/receive timeout, with the same
effect on a peer that has stopped sending. Requests that trip it are retried by the SDK.

The docs for this parameter are corrected too — they described it as a connection-establishment
timeout (that is `object_storage_connect_timeout_ms`) and marked it non-mutable in zh/ja despite
`CONF_mInt64`.

### Measured, on the same cluster and method

| binary | Poco | `object_storage_request_timeout_ms` | queries | hung ~60 s |
|---|---|---|---|---|
| before | on | unset | 522 | **7** |
| before | on | 5 s | 51 | **2** |
| before | off | unset | 429 | 0 |
| before | off | 5 s | 522 | 0 |
| **after** | **on** | **unset** | 153 | **2** |
| **after** | **on** | **5 s** | 517 | **0** |

Row 2 is the bug stated plainly: setting the timeout changed nothing, because it was being
dropped. Row 5 is the fix behaving as specified — with the timeout unset there is nothing to
apply, so the hang is expected to remain, and it does. Row 6 is the fix doing its job: 517
queries across both benchmarks, none hung, worst case 19.9 s against 73.1 s before.

Tests are pure unit tests — Poco connects lazily on `sendRequest()`, so a session for an address
nothing listens on is enough to inspect what was applied to it.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [x] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [x] I have added documentation for my new feature or new function
  - [ ] This pr needs auto generate documentation
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 4.1
  - [x] 4.0
  - [x] 3.5

### CODE REVIEW

**Comment by CelerData-Reviewer** [comment]
@codex review

**Comment by kevincai** [comment]
remove the change from v3.5, turn it off since v4.0+ for now.

**Review by kevincai — DISMISSED** [review]

**Comment by CelerData-Reviewer** [comment]
@codex review

File: be/src/platform/aws/poco_common.cpp [hunk 1]
```diff
@@ -39,6 +39,38 @@ void setTimeouts(Poco::Net::HTTPClientSession& session, const ConnectionTimeouts
...
+    // timeout would let it inherit another client's.
+    //
+    // A non-positive value means "unset": object_storage_request_timeout_ms defaults to -1, which
+    // arrives as a negative Timespan, and Poco gives no defined meaning to that. Unset restores
+    // Poco's default instead of passing the value through.
+    const bool has_timeout =
+            timeouts.send_timeout.totalMicroseconds() > 0 && timeouts.receive_timeout.totalMicroseconds() > 0;
+    const Poco::Timespan& send = has_timeout ? timeouts.send_timeout : poco_default_timeout();
+    const Poco::Timespan& receive = has_timeout ? timeouts.receive_timeout : poco_default_timeout();
+    session.setTimeout(timeouts.connection_timeout, send, receive);
```
Comment (kevincai) [inline be/src/platform/aws/poco_common.cpp]: This gates on `> 0`, so a configured `object_storage_request_timeout_ms=0` is indistinguishable from unset and falls back to `poco_default_timeout()` (Poco's 60 s). That contradicts the semantics this PR itself documents, on the default client path (`enable_poco_client_for_aws_sdk` is `true`):

- `be/src/common/config.h`: "Curl rounds the value down to whole seconds; 0 disables the check, and a negative value leaves the client on its own default."
- `docs/en/administration/configuration/BE_parameters/query_loading.md` (and ja/zh): "`0` disables the check; a negative value leaves the client on its own default, which on the Poco path is 60 seconds."

It also undoes the intent of the pre-existing `>= 0` guard in `fs_s3.cpp`, whose comment states "0 is meaningful for object_storage_request_timeout_ms" -- a deliberate outcome of the review on #25690:

https://github.com/StarRocks/starrocks/blob/1b15119f32e5f6f99ac749a5438ec7a297689b56/be/src/fs/fs_s3.cpp#L175-L179

The curl path does honor 0 (passed straight through as `CURLOPT_LOW_SPEED_TIME=0`), so the two backends now disagree on the documented escape hatch -- which matters more now that the default is no longer unset. Splitting the branch so only a strictly negative value falls back, and `0` is passed through to `setTimeout()`, would match the docs.

Also note no test covers `Poco::Timespan(0)`; `UnsetTimeoutDoesNotInheritThePreviousRequests` exercises only the negative case.

**Comment by CelerData-Reviewer** [comment]
@codex review

File: be/src/platform/aws/poco_common.cpp (hunk 1 above)
Comment (luohaha) [reply be/src/platform/aws/poco_common.cpp]: Fixed in 7b2e4b5f0e1. `apply_request_timeouts()` now restores the Poco default only for strictly negative values and passes zero through to `setTimeout()`, matching the Curl path and the documented semantics. I also added `ZeroTimeoutDisablesThePreviousRequestsTimeout` to cover a pooled session switching from a positive timeout to `Poco::Timespan(0)`.

File: be/src/platform/aws/poco_common.cpp [hunk 2]
```diff
@@ -39,6 +39,38 @@ void setTimeouts(Poco::Net::HTTPClientSession& session, const ConnectionTimeouts
...
+    // for, never in whatever state the previous borrower left behind -- skipping the call when a
+    // request has no timeout would let it inherit another client's.
+    //
+    // A negative value means "unset" and Poco gives no defined meaning to a negative Timespan, so
+    // restore Poco's default in that case. Zero is meaningful: it disables the timeout and must be
+    // passed through just like it is on the Curl path.
+    const Poco::Timespan& send =
+            timeouts.send_timeout.totalMicroseconds() >= 0 ? timeouts.send_timeout : poco_default_timeout();
+    const Poco::Timespan& receive =
+            timeouts.receive_timeout.totalMicroseconds() >= 0 ? timeouts.receive_timeout : poco_default_timeout();
```
[the remark this answers is not in the document]
Comment (luohaha) [reply be/src/platform/aws/poco_common.cpp:66]: Fixed in 301386cf84f. Both direct S3 configuration paths now preserve an unset value as a negative sentinel when the Poco client is enabled, while an explicit zero remains zero. Starlet startup and runtime updates apply the same tri-state mapping (`-1` for Poco, SDK-default `0` for Curl). The new tests cover the real `TCloudConfiguration` path and the Starlet mapping.

File: be/src/platform/aws/poco_common.cpp [hunk 3]
```diff
@@ -119,6 +151,11 @@ PooledHTTPSessionPtr HTTPSessionPools::getSession(const Poco::URI& uri, const Co
     }
 
     auto session = pool->get(timeouts.connection_timeout.totalMicroseconds());
+    // A pooled session is reused, so whatever the previous caller left on it is still in
+    // effect -- apply this request's timeouts before handing it out. Without this the send and
+    // receive timeouts computed from ClientConfiguration were dropped here, and a read that
+    // stopped receiving waited out Poco's built-in default instead of the configured value.
+    apply_request_timeouts(*session, timeouts);
```
[the remark this answers is not in the document]
Comment (luohaha) [reply be/src/platform/aws/poco_common.cpp:158]: Fixed in 301386cf84f. `requestTimeoutMs` is now part of the StarRocks S3 client-cache identity, with a regression test asserting that ordinary 10 s and rename 30 s configurations create distinct clients. The corresponding Starlet cache fix is tracked in StarRocks/staros#1328.

**Comment by luohaha** [comment]
@codex review

**Review by xiangguangyxg — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Review by starrocks-xupeng — APPROVED** [review]
